src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -68,116 +78,196 @@ private static VersionInfo GetVersionInfo()
+            }
+            (_inputTypes, _outputTypes) = GetTypes(input.Schema);
+        }
+
+        private NAIndicatorTransform(IHost host, ModelLoadContext ctx)
+            : base(host, ctx)
+        {
+            Host.AssertValue(ctx);
+            _outputTypes = null;
+            _inputTypes = null;
         }
 
         /// <summary>
-        /// Public constructor corresponding to SignatureDataTransform.
+        /// Factory method for SignatureDataTransform.
         /// </summary>
-        public NAIndicatorTransform(IHostEnvironment env, Arguments args, IDataView input)
-            : base(env, RegistrationName, Contracts.CheckRef(args, nameof(args)).Column,
-                input, TestType)
+        public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
         {
-            Host.AssertNonEmpty(Infos);
-            Host.Assert(Infos.Length == Utils.Size(args.Column));
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
+            env.CheckValue(input, nameof(input));
+
+            env.CheckValue(args.Column, nameof(args.Column));
+            var cols = new ColumnInfo[args.Column.Length];
+            for (int i = 0; i < cols.Length; i++)
+            {
+                var item = args.Column[i];
 
-            _types = GetTypesAndMetadata();
+                cols[i] = new ColumnInfo(item.Source, item.Name);
+            };
+            return new NAIndicatorTransform(env, input, cols).MakeDataTransform(input);
         }
 
-        private NAIndicatorTransform(IHost host, ModelLoadContext ctx, IDataView input)
-            : base(host, ctx, input, TestType)
+        /// <summary>
+        /// Factory method for SignatureDataTransform.
+        /// </summary>
+        public static NAIndicatorTransform Create(IHostEnvironment env, ModelLoadContext ctx)
         {
-            Host.AssertValue(ctx);
-            Host.AssertNonEmpty(Infos);
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register(RegistrationName);
+
+            host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            return new NAIndicatorTransform(host, ctx);
+        }
+
+        /// <summary>
+        /// Factory method for SignatureDataTransform.
+        /// </summary>
+        public static IDataTransform Create(IHostEnvironment env, IDataView input, params ColumnInfo[] columns)
+            => new NAIndicatorTransform(env, input, columns).MakeDataTransform(input);
 
-            // *** Binary format ***
-            // <base>
+        /// <summary>
+        /// Factory method for SignatureDataTransform.
+        /// </summary>
+        public static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+            => Create(env, ctx).MakeDataTransform(input);
+
+        /// <summary>
+        /// Factory method for SignatureDataTransform.
```

Comment:
>SignatureDataTransform [](start = 31, length = 22)

Please pay attention to the comments you write :) They should be correct. This one, for example, is a factory for `SignatureLoadRowMapper`. And also private